pkg/app/piped/executor/ecs/ecs.go

```diff
@@ -451,6 +451,19 @@ func routing(ctx context.Context, in *executor.Input, platformProviderName strin
 		return false
 	}
 
+	currListenerRuleArns, err := client.GetListenerRuleArns(ctx, currListenerArns)
+	if err != nil {
+		in.LogPersister.Errorf("Failed to get current active listener rule: %v", err)
+		return false
+	}
+
+	if len(currListenerRuleArns) > 0 {
+		if err := client.ModifyRules(ctx, currListenerRuleArns, routingTrafficCfg); err != nil {
+			in.LogPersister.Errorf("Failed to routing traffic to PRIMARY/CANARY variants: %v", err)
+			return false
+		}
```

Comment:
Same as below, we should return after this line

Reply:
[abd12ca](https://github.com/pipe-cd/pipecd/pull/4668/commits/abd12ca3edf73067525c6bf9be90757b91938485)
Thank you,Correspondence🙏